daemon/container/container.go

```diff
@@ -647,7 +647,10 @@ func (container *Container) BackfillEmptyPBs() {
 	if container.HostConfig == nil {
 		return
 	}
-
+	if container.HostConfig.PortBindings == nil {
+		container.HostConfig.PortBindings = networktypes.PortMap{}
+		return
+	}
```

Comment:
Per discussion on Slack with @robmry - also adding an extra safeguard here to make sure we don't end up with a nil map.